### PR TITLE
Save/Record, Program Info and Logged Data features in Focus Mode

### LINKED ISSUE #344: Record/save feature in Focus Mode
**Actual Behaviour.**  
Missing Record/save feature in Focus Mode.

**Expected Behaviour**  
Record/save feature in Focus Mode.

**Steps to reproduce it**  
Go to Focus mode, observe the toolbar.

**Additional context**  
> Not available

**Would you like to work on the issue?**
Yes.

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #344

**Changes**: Save/Record, Program Info and Logged Data features in Focus Mode.

**Screenshot/s for the changes**: 

 

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[app-debug.zip](https://github.com/fossasia/neurolab-android/files/3361263/app-debug.zip)

### CODE REVIEW

**Comment by jddeep** [comment]
@CloudyPadmal Codacy is complaining about some unused config parameters. Should I then remove them for now?

**Comment by CloudyPadmal** [comment]
Yes we can remove or if we might need them in future, we can comment them out for now

**Comment by jddeep** [comment]
> Yes we can remove or if we might need them in future, we can comment them out for now

@CloudyPadmal Cool, I am converting them to local for now. :+1:

**Review by mariobehling — CHANGES_REQUESTED** [review]
Instead of a dropdown menu, please implement two buttons Record and a Stop button: When pressed, data will be recorded from that time onwards. When pressed Stop, the recording will be stopped and data will be automatically stored in the Datalogger. Please implement it as two buttons.

**Comment by jddeep** [comment]
Done. Thanks.

File: app/src/main/java/io/neurolab/main/output/visual/SpaceAnimationVisuals.java [hunk 1]
```diff
@@ -35,14 +42,46 @@ public static void spaceAnim(View view) {
         handleIndicator(view);
     }
 
+    public static void stopAnim() {
+        try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
```
Comment (CloudyPadmal) [inline app/src/main/java/io/neurolab/main/output/visual/SpaceAnimationVisuals.java]: Let's upgrade project minSDK to 23. Lower versions have many problems with OTG. What's the version of your device?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -107,6 +107,10 @@ A difference to existing projects like OpenBCI is that it will not be necessary
     <string name="freq_label">Frequency (Hertz)</string>
     <string name="eeg_data_label">EEG data (mV)</string>
     <string name="play_graph">Play Graph</string>
+    <string name="save_focus_data">Save Focus Data</string>
+    <string name="focus_program_info">This Program Mode helps you increase you increase your focus and concentration levels.</string>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: Check

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/neurolab/main/output/visual/SpaceAnimationVisuals.java (hunk 1 above)
Comment (jddeep) [reply app/src/main/java/io/neurolab/main/output/visual/SpaceAnimationVisuals.java]: Android 8.0.0

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (jddeep) [reply app/src/main/res/values/strings.xml]: Ah! Typos :sweat_smile: Thanks

File: app/src/main/java/io/neurolab/main/output/visual/SpaceAnimationVisuals.java (hunk 1 above)
Comment (CloudyPadmal) [reply app/src/main/java/io/neurolab/main/output/visual/SpaceAnimationVisuals.java]: Then no problem for you to work with right? Let's upgrade minSDK version here.

Comment (jddeep) [reply app/src/main/java/io/neurolab/main/output/visual/SpaceAnimationVisuals.java]: Yup. Will do.

**Comment by mariobehling** [comment]
Please also resolve conflicts.

**Comment by jddeep** [comment]
@CloudyPadmal Change made. Thanks.

**Review by CloudyPadmal — APPROVED** [review]

**Review by mariobehling — APPROVED** [review]

**Pull request merged**